### PR TITLE
4.17 - KVM upgrade issue CPU speed detection

### PR DESCRIPTION
**Pull request by nvazquez**
Including docs from https://github.com/apache/cloudstack/issues/6680 reported issue and investigation

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan docbuild

File: source/upgrading/upgrade/mysql.rst [hunk 1]
```diff
@@ -24,7 +24,8 @@ not be able to start any VM.
 The following SQL statement needs to be manually executed in order to fix such issue:
 
    .. parsed-literal::
-ALTER TABLE nics MODIFY COLUMN update_time timestamp DEFAULT CURRENT_TIMESTAMP;
+
+      ALTER TABLE nics MODIFY COLUMN update_time timestamp DEFAULT CURRENT_TIMESTAMP;
```
Comment (DaanHoogland) [inline source/upgrading/upgrade/mysql.rst:28]: is this change needed?

**Review by DaanHoogland — COMMENTED** [review]
text looks good, but ... is this only an issue on upgraded machines? It seem to be an issue on all amd/ubuntu based machines.
Can the problem be solved in code e.g.
- add a last try to read the CPU speed from that property
- add a hook for a script that can do whatever the operator thinks is sensible to return the speed
- ...
?

File: source/upgrading/upgrade/kvm-417.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+
+KVM agents CPU speed detection issue
+====================================
+
+There is an issue with AMD (and possibly other CPU architectures) on the CPU speed detection at the 
+4.17.0 and 4.17.0.1 CloudStack agents on KVM hosts at startup. The CloudStack agents will attempt to 
+get the CPU speed  by reading the `/sys/devices/system/cpu/cpu0/cpufreq/base_frequency` information.
```
Comment (weizhouapache) [inline source/upgrading/upgrade/kvm-417.rst:21]: cloudstack agent tries to get cpu speed in different ways
see https://github.com/apache/cloudstack/pull/5506

I personally think it is a libvirt issue, `virsh nodeinfo` returns incorrect value with AMD cpus.

see https://github.com/apache/cloudstack/issues/6680#issuecomment-1234492354

File: source/upgrading/upgrade/kvm-417.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+   specific language governing permissions and limitations
+   under the License.
+
+KVM agents CPU speed detection issue
+====================================
+
+There is an issue with AMD (and possibly other CPU architectures) on the CPU speed detection at the 
+4.17.0 and 4.17.0.1 CloudStack agents on KVM hosts at startup. The CloudStack agents will attempt to 
+get the CPU speed  by reading the `/sys/devices/system/cpu/cpu0/cpufreq/base_frequency` information. 
+However, the CPU speed information is not included in all the CPU architectures.
```
Comment (shwstppr) [inline source/upgrading/upgrade/kvm-417.rst:22]:
```suggestion
However, the CPU speed information is not included in all the CPU architectures. In such cases, `libvirt` is used to retrieve the CPU speed which may not always report the correct base speed value for the CPU.
```

**Review by shwstppr — COMMENTED** [review]
Mostly okay but it would be good if we can give complete detail on how the value is retrieved so the operator doesn't get confused in cases when libvirt value is used.

**Comment by shwstppr** [comment]
Fixed with code PR, https://github.com/apache/cloudstack/pull/6696
@nvazquez cc @DaanHoogland please re-open if needed
